### PR TITLE
fix(client):unified the dataset table name when copy

### LINKED ISSUE #1731: Unified the dataset table name when copy
**Describe the bug**
It will have different table names when copy dataset to cloud use bellow commands: 
- swcli dataset copy ucf101/version/latest cloud://e2e0128/project/1(table name is 'project/1/...')
- swcli dataset copy ucf101/version/latest cloud://e2e0128/project/starwhale(table name is 'project/starwhale/...')
But it is unified on the cloud with the same name 'project/starwhale/...'.
And it will lead error when dataset load data: `ai.starwhale.mlops.exception.SwValidationException: invalid request on subject Starwhale Internal DataStore
invalid table name project/starwhale/dataset/ucf101/7m/7mguzpokk7ez3xssi4l3rk4zhesd6we3dvyj4aot/meta`,and the reason is that the real name is 'project/1/dataset/ucf101/7m/7mguzpokk7ez3xssi4l3rk4zhesd6we3dvyj4aot/meta'

image
image
image
image

**To Reproduce**

**Expected behavior**

**Environment**

### PR DESCRIPTION
**Pull request by goldenxinxing**
## Description

closes #1731 

1. origin copy:
image
2. changed:
image

## Modules
- [ ] UI
- [ ] Controller
- [ ] Agent
- [x] Client
- [ ] Python-SDK
- [ ] Others

## Checklist
- [ ] run code format and lint check
- [x] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: client/starwhale/core/dataset/copy.py [hunk 1]
```diff
@@ -141,3 +149,18 @@ def _do_download_bundle_dir(self, progress: Progress) -> None:
                 local._info = copy.deepcopy(remote.info)
 
         super()._do_download_bundle_dir(progress)
+
+    def _get_remote_project_name(self, instance: URI, project: str) -> Any:
+        resp = self.do_http_request(
+            f"/project/{project}",
+            instance_uri=instance,
+            method=HTTPMethod.GET,
+            use_raise=True,
```
Comment (tianweidut) [inline client/starwhale/core/dataset/copy.py:158]: if `use_raise=True`, the http request will check the status code automatically, so can we ignore line160-161?

File: client/starwhale/core/dataset/copy.py [hunk 2]
```diff
@@ -141,3 +149,18 @@ def _do_download_bundle_dir(self, progress: Progress) -> None:
...
+        )
+        if resp.status_code != HTTPStatus.OK:
+            raise NotFoundError(f"project:{project}")
+
+        resp_body = resp.json()
+
+        _project = resp_body.get("data")
```
Comment (tianweidut) [inline client/starwhale/core/dataset/copy.py:165]:
```
        _project = resp_body.get("data")
        return _project["name"] if _project else None
```
-->
```
        return resp_body.get("data", {}).get("name")
```

**Comment by tianweidut** [comment]
@goldenxinxing 
1. use another meaningful tile for this pr?
2. should we consider model/runtime copy the same issue?

File: client/starwhale/core/dataset/copy.py (hunk 1 above)
Comment (goldenxinxing) [reply client/starwhale/core/dataset/copy.py:158]: done

File: client/starwhale/core/dataset/copy.py (hunk 2 above)
Comment (goldenxinxing) [reply client/starwhale/core/dataset/copy.py]: done

**Comment by goldenxinxing** [comment]
> @goldenxinxing
> 
> 1. use another meaningful tile for this pr?
> 2. should we consider model/runtime copy the same issue?

Copy model and runtime doesn't have this problem, this error only occur on dataset copy. #1731

**Review by tianweidut — APPROVED** [review]
LGTM

**Pull request merged**